### DISCUSSION TITLE
Feature request: AsciiDoc support for GitHub Pages

### DISCUSSION THREAD

**Discussion by laubai** [community/community#36737] [Other Feature Feedback, Questions, & Ideas]
I'm sure I'm not the only one who'd like support for asciidoc as well as markdown as a source for GitHub Pages.

If you have recommendations for how to work around the absence of this support in the meantime, I'd love to hear why you recommend them. :)

**Comment by ossanna16** [comment]
Hey there @laubai and thank you for the feature request! I'm glad you found our community 🙂

**Comment by yoannchaudet** [comment]
[Jekyll](https://jekyllrb.com/) which Pages natively supports (i.e. if you enable Pages to track a given branch in a repository, it will build with Jekyll) does understand Markdown already.

Staying in the Jekyll ecosystem, there is also this plugin for AsciiDoc which you may find helpful: https://github.com/asciidoctor/jekyll-asciidoc. To use this plugin though you will need your own workflow for building and deploying a Pages site. We have an example of such a workflow [here](https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml).

**Reply by ericzolf** [reply]
GitHub else supports asciidoc as a format for all kinds of documentation, why not in GitHub pages? Could it be added to the [supported plugins](https://pages.github.com/versions/), instead of pushing us to workarounds?

**Reply by ericzolf** [reply]
I'm now using my own instance of Jekyll but it takes a lot longer to install, so it uses more resources than if I could use the standard Jekyll, so there is perhaps an ROI?

**Reply by Tejeev** [reply]
There are so many flavors of md out there that I had to chose one, and I ended up choosing ASCIIDOC instead because it's standardized, functional, simple, and fairly ubiquitious. While github has had a hand in making one particular branch of md more popular than others, we can't say it's at all ubiquitous given how fractious the markdown ecosystem is.  As such, I don't think we can generalize by saying that "Jekyll supports markdown" as it supports only a given flavor of md.   That may be a touch pedantic but I think it matters because ASCIIDOC is a very strong competitor for a standardized markdown which is human readable and feature rich.  More importantly, everywhere outside of Github I see more adoption of ASCIIDOC.  

Given the above, I think it makes sense to request the addition of a jekyll-asciidoc to the supported plugins, and I think that is what this feature request is specifically asking for.

**Comment by Tejeev** [comment]
I also wish for the addition of asciidoc support.  Perhaps through the addition of [jekyll-asciidoc](https://github.com/asciidoctor/jekyll-asciidoc) to the [supported plugins](https://pages.github.com/versions/).
